### ISSUE TITLE
Store Functionality

### ISSUE THREAD

**Issue by Maxisthemoose** [Maxisthemoose/tycoon-bot#6]
## Stores
Stores should be able to be bought in some form of menu. Each store has a given output. This output will output that stores designated item. Then every time the user "checks in", they can sell the amount of items that were created in the store for cash. I can only really think of one way to do this right now. The user object can have a property on them which will store when their last "check-in" with the bot was, then take the current time, and figure out how long the user was idle. Then we can take that number and multiply it by each stores output, and those are the items that they have to sell. They can then sell these and the "check-in" value will be reset to the current time. The total output of a store can be modified by user bought boosters and / or prestige levels.

#### If you have any thoughts or a better way of doing this, comment bellow.

**Comment by CreeperPlanet26** [comment]
I like how this works.

**Comment by Maxisthemoose** [comment]
## Minimum Idle Time
I feel like we should make a minimum amount of time the user has to be away before they can collect and sell all their goods. I was thinking somewhere in the range of 30m to 1h. As we can make some mini games that the user can play yo distract themselves and earn bonus money and maybe even some power ups of some sort.

## Maximum Idle Time
There should **definitely** at least be a max idle time. This would just make it so the user will be more likely to come back to the bot and continue playing

**Comment by CreeperPlanet26** [comment]
Yeah like a shop can be full. But I think min should be 5 mins cuz that's how long it takes to make a sale.

**Comment by ImBoing** [comment]
Wait this is to figure out how afk stuff will work right? Because I was thinking we could do that but in order to create your own shop you do something like a found command and you give your store a name then you can start working and doing whatever.

**Comment by CreeperPlanet26** [comment]
Yeah this is selling

**Comment by ImBoing** [comment]
Oh alright I was also thinking to be more realistic we could have customer happiness percentage. Like lets say every 12 hours you need to clean your shop, and if you don't clean it every 12 hours your happiness percentage goes down and you don't make as much money when working.

**Comment by CreeperPlanet26** [comment]
I see so the more workers you have the more you have to clean.

**Comment by ImBoing** [comment]
> I see so the more workers you have the more you have to clean.

I mean thats a possibility but idk if it makes sense, maybe as you level up you have to clean more or the less you clean the faster that percentage drops?

**Comment by CreeperPlanet26** [comment]
yeah sounds like a good idea but we should see what duck thinks

**Comment by Maxisthemoose** [comment]
Totally fine by me

**Comment by CreeperPlanet26** [comment]
bet i think we should do something like that then

**Comment by Maxisthemoose** [comment]
Sounds good

**Comment by Maxisthemoose** [comment]
We may want to put the check in time on the store object itself in case someone buys a shop like halfway through their "away timer", it will act like a normal shop bought before the interaction since the user made the interaction before they bought the store

**Comment by CreeperPlanet26** [comment]
How many stores can a user have?

**Comment by CreeperPlanet26** [comment]
I think a user should have 1 and there will be a item shop where u can buy boosters and boxes.
